openquakeplatform_ipt/static/ipt/js/earthquake_rupture.js

```diff
@@ -0,0 +1,626 @@
+                var middle_table_id = er_obj.pfx + 'div[name="rupture"] > div[name="complex"] div[name="complex-' + id + '"] div[name="middles"] div[name="middle-geometry-' + id + '-' + mid_id + '"]';
+                $(middle_table_id).remove();
+                delete(this.complex_tbl[id].middles[mid_id]);
+            }
+        }
+        delete(this.complex_tbl[id]);
+        $item.remove();
+    },
+
+    complex_surface_add: function () {
+        var ct = er_obj.complex_tbl_cur;
+        var ctx = '\
+      <div name="complex-' + ct + '">\n\
+        <div class="menuItems" style="margin-top: 12px; margin-left: 100px;">\n\
+            <div style="display: inline-block; float: left;"><h4>Complex surface ' + (ct + 1) + '</h4></div>';
+
+        ctx += (ct == 0 ? '<div style="clear: both;"></div>' : '            <button type="button" data_gem_id="' + ct + '" class="btn" style="margin-top: 8px; margin-bottom: 8px;" onclick="er_obj.complex_surface_del(this);">Delete Complex Surface</button>');
+        ctx += '\n\
+        </div>\n\
+        <div class="menuItems complex_geometry">\n\
+          <label>Fault top edge:</label>\n\
+          <div style="margin-left: auto;">\n\
+            <div name="top-geometry-' + ct + '" style="margin-left: auto; width: 240px; height: 90px; overflow: hidden;"></div>\n\
+          </div>\n\
+        </div>\n\
+        <div name="middles"></div>\n\
+        <div class="menuItems" style="margin-top: 12px; text-align: center;">\n\
+          <button type="button" data_gem_id="' + ct + '" class="btn" style="margin-top: 8px; margin-bottom: 8px;" onclick="er_obj.complex_surface_middle_add(this);">Add intermediate edge</button>\n\
+        </div>\n\
+        <div class="menuItems complex_geometry">\n\
+          <label>Fault bottom edge:</label>\n\
+          <div style="margin-left: auto;">\n\
+            <div name="bottom-geometry-' + ct + '" style="margin-left: auto; width: 240px; height: 90px; overflow: hidden;"></div>\n\
+          </div>\n\
+        </div>\n\
+</div>';
+        $(er_obj.pfx + 'div[name="complexes"]').append(ctx);
+
+        er_obj.complex_tbl[ct] = { top: null, bottom: null, middles: {}, middles_cur: 0 };
+
+        var top_table_id = er_obj.pfx + 'div[name="rupture"] > div[name="complex"] div[name="top-geometry-' + ct + '"]';
+
+        $(top_table_id).handsontable(er_obj.tbl_complex_params);
+        er_obj.complex_tbl[ct].top = $(top_table_id).handsontable('getInstance');
+
+        var bottom_table_id = er_obj.pfx + 'div[name="rupture"] > div[name="complex"] div[name="bottom-geometry-' + ct + '"]';
+        $(bottom_table_id).handsontable(er_obj.tbl_complex_params);
+        er_obj.complex_tbl[ct].bottom = $(bottom_table_id).handsontable('getInstance');
+
+        er_obj.complex_tbl_cur++;
+    },
+
+    /***************
+     *             *
+     *  ARBITRARY  *
+     *             *
+     ***************/
+
+    arbitrary_geometry_populate: function(obj) {
+        var thiz = this;
```

Comment:
Ad a comment explain that thiz is a closure variable for $.post() associated callbacks.